File: tutopy_pkg/tutopy_pkg/publisher_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String


class PublisherNode(Node):
    def __init__(self):
        super().__init__("publisher_node")
        self.publisher_ = self.create_publisher(String, "first_publisher", 10)
        self.timer_ = self.create_timer(0.5, self.publish)
        self.get_logger().info("Publisher Noder has been started")

    def publish(self):
        msg = String()
        msg.data = "Hi Python"
        self.publisher_.publish(msg)


def main(args=None):
    rclpy.init(args=args)

    node = PublisherNode()

    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: tutocpp_pkg/src/publisher_node.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int16.hpp"

class PublisherNode : public rclcpp::Node
{
public:
    PublisherNode() : Node("publisherCpp")
    {
        publisher_ = create_publisher<example_interfaces::msg::Int16>("number", 10);
        timer = create_wall_timer(std::chrono::seconds(1), std::bind(&PublisherNode::publish, this));
        RCLCPP_INFO(get_logger(), "Publisher cpp Node has been started");
    }

private:
    rclcpp::Publisher<example_interfaces::msg::Int16>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer;

    void publish()
    {
        auto msg = example_interfaces::msg::Int16();
        msg.data = 2;
        publisher_->publish(msg);
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<PublisherNode>();

    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}